guides/faq.rst

```diff
@@ -343,6 +343,14 @@ with a lot of work).
 
 Always back up all your files!
 
+
+Why shouldn't I use underscores in my device name?
+--------------------------------------------------
+
+The top level ``name:`` field in your .yaml file defines the node name(/hostname) on the local network.  According to `RFC1912 <https://datatracker.ietf.org/doc/html/rfc1912>`_, underscore characters (``_``) in hostnames are not valid.  In reality some local DNS/DHCP setups will be ok with underscores and some will not.  If connecting via a static IP address, there will probably be no issues.  In some cases, initial setup using an underscore works, but then the connection fails when Home Assistant restarts.  Recommendation: use hyphen (``-``) instead of underscore if you can.  
+
+Important: follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to use the ``use_address`` parameter when renaming a live device, as the connection to an existing device will only work with the old name until the name change is complete.
```

Comment:
```suggestion
Follow these `instructions </components/esphome.html#changing-esphome-node-name>`_ to change the underscores in the name to hyphens.
```

No need to explain why here, it's explained in the instructions.

Reply:
I think it is better to give some idea why here.  Many will assume they can just rename it.